### PR TITLE
Feat/token registry accept surrender

### PR DESCRIPTION
**Pull request by justussoh**
This PR adds the ability to shred documents for token registry minters.

Refer to story for documents:
https://www.pivotaltracker.com/story/show/171761567

### CODE REVIEW

File: src/components/AssetManagementPanel/AssetManagementApplication/index.tsx [hunk 1]
```diff
@@ -24,20 +29,37 @@ export const AssetManagementApplication = ({
...
     transferToState,
+    acceptSurrenderingState,
+    destroyToken,
     isSurrendered,
     isTitleEscrow,
     approveNewTransferTargets,
     approveNewTransferTargetsState,
     transferToNewEscrow,
     transferToNewEscrowState,
+    titleEscrowOwner,
```
Comment (justussoh) [inline src/components/AssetManagementPanel/AssetManagementApplication/index.tsx]: Rename to `documentOwner`

File: src/common/hooks/useTitleEscrowContract.ts [hunk 2]
```diff
@@ -21,13 +20,18 @@ export const useTitleEscrowContract = (
     setTitleEscrow(instance);
   }, [provider, tokenId, tokenRegistry]);
 
+  const reset = useCallback(() => {
```
Comment (justussoh) [inline src/common/hooks/useTitleEscrowContract.ts]: Implement reset in tokenContext

File: src/components/AssetManagementPanel/AssetManagementApplication/index.tsx [hunk 3]
```diff
@@ -24,20 +29,37 @@ export const AssetManagementApplication = ({
...
   const [assetManagementAction, setAssetManagementAction] = useState(AssetManagementActions.None);
-  const { upgradeProvider, account } = useProviderContext();
+  const { upgradeProvider, account, provider } = useProviderContext();
+  const { tokenRegistry } = useTokenRegistryContract(tokenRegistryAddress, provider);
+  // Check if direct owner is minter
+  const { call: checkIsMinter, value: isMinter } = useContractFunctionHook(tokenRegistry, "isMinter");
+
+  useEffect(() => {
+    if (isTitleEscrow === false && account) {
+      checkIsMinter(account);
```
Comment (justussoh) [inline src/components/AssetManagementPanel/AssetManagementApplication/index.tsx:51]: Add a comment on the return result to me ?.[0]

File: src/components/AssetManagementPanel/AssetManagementForm/AssetManagementForm.tsx [hunk 4]
```diff
@@ -57,21 +66,29 @@ export const AssetManagementForm = ({
...
   isTitleEscrow,
 }: AssetManagementFormProps) => {
-  const isHolder = account === holder;
-  const isBeneficiary = account === beneficiary;
+  const isHolder = isTitleEscrow && account === holder;
+  const isBeneficiary = isTitleEscrow && account === beneficiary;
   const canSurrender = isBeneficiary && isHolder;
-  const canEndorseBeneficiary = isBeneficiary && isHolder;
-  const canNominateBeneficiaryHolder = isBeneficiary && !isHolder;
+  /* Change here is tokenRegistryAdmin,
```
Comment (justussoh) [inline src/components/AssetManagementPanel/AssetManagementForm/AssetManagementForm.tsx]: Phrase properly

File: src/test/fixture/ebl-accepted surrender.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (simboonlong) [inline src/test/fixture/ebl-accepted surrender.json]: missing dash `ebl-accepted-surrender.json`

**Comment by simboonlong** [comment]
lgtm

start surrender
Screenshot 2020-10-01 at 1 53 02 PM

after surrendered
Screenshot 2020-10-01 at 2 41 41 PM

switch account to the one that deployed the token registry, which this document was issued to
Screenshot 2020-10-01 at 2 41 57 PM

start shred (burn token)
Screenshot 2020-10-01 at 2 42 02 PM

after shred (burnt => 0x00...dEaD)
Screenshot 2020-10-01 at 2 42 26 PM

**Review by simboonlong — APPROVED** [review]

**Pull request merged**